### ISSUE TITLE
:CocCommand latex.UpdateLanguageServer doesn't download new server

### ISSUE THREAD

**Issue by oblitum** [fannheyward/coc-texlab#32]
```
~/.config/coc/extensions/coc-texlab-data
[I] ➜ ls -lh
total 19M
-rwxr-xr-x 1 francisco francisco 19M Jul 13 10:01 texlab
```

I have this old binary on `~/.config/coc/extensions/coc-texlab-data`. I executed `:CocCommand latex.UpdateLanguageServer` but it didn't download a new binary.

**Comment by fannheyward** [comment]
<img width="351" alt="image" src="https://user-images.githubusercontent.com/345274/62597358-59e45280-b918-11e9-966b-da05a6929515.png">

<img width="425" alt="image" src="https://user-images.githubusercontent.com/345274/62597377-67014180-b918-11e9-9855-f702b1884d56.png">

After the command executed, you will see `Downloading Texlab Server` on status line, and then shows download progress percent and version.

**Comment by oblitum** [comment]
Nothing of that happens. I checked whether it was downloading with `speedometer`. It didn't start.

**Comment by oblitum** [comment]
`./texlab -V` for the old bin returned `1.1.0`,  the new one which I've downloaded manually returns `1.3.0`. The request for latest on github also returns `1.3.0`.

**Comment by fannheyward** [comment]
I just published v1.8.1, which added more info of downloading.

1. `Getting the latest version...` is fetching from https://api.github.com/repos/latex-lsp/texlab/releases/latest , parse to get the latest version tag_name, `v1.3.0`.

<img width="421" alt="image" src="https://user-images.githubusercontent.com/345274/62597649-50a7b580-b919-11e9-8207-158965ab9cdc.png">

2. Start to download, with progress percent.

<img width="424" alt="image" src="https://user-images.githubusercontent.com/345274/62597913-1ee31e80-b91a-11e9-8384-d770f6895e5f.png">

Try `:CocUpdate`, and then `:CocCommand latex.UpdateLanguageServe`.

**Comment by oblitum** [comment]
Sorry but this progress simply doesn't show for me, no downloads...

**Comment by fannheyward** [comment]
Are you using a proxy? I mean `http.proxy` in `coc-settings.json`?

**Comment by oblitum** [comment]
No.

**Comment by oblitum** [comment]
I did a GET with `curl` on the latest URL, it returns fine, dunno why the plugin isn't doing it, I'm trying to figure why.

**Comment by fannheyward** [comment]
[index.js.zip](https://github.com/fannheyward/coc-texlab/files/3475532/index.js.zip)

download and unzip, put `index.js` to `~/.config/coc/extensions/node_modules/coc-texlab/lib`.

1. `:CocCommand latex.UpdateLanguageServe`
2. `:CocOpenLog`

you will see logs like this:

```
 15 2019-08-07T14:15:45.302 ERROR (pid:99642) [extension-coc-texlab] - getting latest version
 16 2019-08-07T14:15:45.302 ERROR (pid:99642) [extension-coc-texlab] - got proxy agent: undefined
 17 2019-08-07T14:15:48.294 ERROR (pid:99642) [extension-coc-texlab] - got latest version v1.3.0
 18 2019-08-07T14:15:51.289 ERROR (pid:99642) [extension-coc-texlab] - downloading 0
 19 2019-08-07T14:15:51.572 ERROR (pid:99642) [extension-coc-texlab] - downloading 0
 20 2019-08-07T14:15:51.581 ERROR (pid:99642) [extension-coc-texlab] - downloading 0
 21 2019-08-07T14:15:51.582 ERROR (pid:99642) [extension-coc-texlab] - downloading 0
 22 2019-08-07T14:15:51.582 ERROR (pid:99642) [extension-coc-texlab] - downloading 0
 23 2019-08-07T14:15:51.867 ERROR (pid:99642) [extension-coc-texlab] - downloading 1
 24 2019-08-07T14:15:51.868 ERROR (pid:99642) [extension-coc-texlab] - downloading 1
```

**Comment by oblitum** [comment]
I'm doing it already, I changed `index.js` locally, it's weird, the download goes from 0% to 1% and then I don't see progress anymore. not sure whether I need to put more logs. The code sticking around `.on('downloadProgress'`, goes from 0% to 1%, I don't see any more progress, and it never reaches `.on('end'` or `.on('error'`.

**Comment by oblitum** [comment]
A nevermind... p was not multiplied by 100 in my logs, let change it.

**Comment by oblitum** [comment]
Ah no, it's not needed in fact.

**Comment by oblitum** [comment]
Same thing with yourr `index.js`:

```
2019-08-07T03:35:43.520 INFO (pid:18034) [services] - registed service "languageserver.reason"
2019-08-07T03:35:43.521 INFO (pid:18034) [services] - registed service "languageserver.ccls"
2019-08-07T03:35:43.521 INFO (pid:18034) [services] - registed service "languageserver.golang"
2019-08-07T03:35:43.521 INFO (pid:18034) [services] - registed service "languageserver.rust"
2019-08-07T03:35:43.521 INFO (pid:18034) [services] - registed service "languageserver.purescript"
2019-08-07T03:35:43.521 INFO (pid:18034) [services] - registed service "languageserver.bash"
2019-08-07T03:35:43.558 INFO (pid:18034) [services] - registed service "eslint"
2019-08-07T03:35:43.699 INFO (pid:18034) [services] - registed service "texlab"
2019-08-07T03:35:43.699 INFO (pid:18034) [services] - TexLab state change: stopped => starting
2019-08-07T03:35:43.784 INFO (pid:18034) [language-client-index] - texlab started with 18051
2019-08-07T03:35:43.804 INFO (pid:18034) [plugin] - coc 0.0.73 initialized with node: v11.15.0
2019-08-07T03:35:43.808 INFO (pid:18034) [services] - TexLab state change: starting => running
2019-08-07T03:35:43.812 INFO (pid:18034) [services] - service texlab started
2019-08-07T03:35:43.828 INFO (pid:18034) [watchman] - watchman watching project: /home/francisco/Projects/invoice
2019-08-07T03:35:43.963 INFO (pid:18034) [watchman] - watchman watching project: /home/francisco/.config/coc/ultisnips
2019-08-07T03:35:52.394 ERROR (pid:18034) [extension-coc-texlab] - getting latest version
2019-08-07T03:35:52.396 ERROR (pid:18034) [extension-coc-texlab] - got proxy agent: undefined
2019-08-07T03:35:53.326 ERROR (pid:18034) [extension-coc-texlab] - got latest version v1.3.0
2019-08-07T03:35:55.257 ERROR (pid:18034) [extension-coc-texlab] - downloading 0
2019-08-07T03:35:55.274 ERROR (pid:18034) [extension-coc-texlab] - downloading 0
2019-08-07T03:35:55.468 ERROR (pid:18034) [extension-coc-texlab] - downloading 0
2019-08-07T03:35:55.471 ERROR (pid:18034) [extension-coc-texlab] - downloading 0
2019-08-07T03:35:55.504 ERROR (pid:18034) [extension-coc-texlab] - downloading 0
2019-08-07T03:35:55.507 ERROR (pid:18034) [extension-coc-texlab] - downloading 0
2019-08-07T03:35:55.667 ERROR (pid:18034) [extension-coc-texlab] - downloading 1
2019-08-07T03:35:55.670 ERROR (pid:18034) [extension-coc-texlab] - downloading 1
2019-08-07T03:35:55.705 ERROR (pid:18034) [extension-coc-texlab] - downloading 1
2019-08-07T03:35:55.706 ERROR (pid:18034) [extension-coc-texlab] - downloading 1
2019-08-07T03:35:55.741 ERROR (pid:18034) [extension-coc-texlab] - downloading 1
```

**Comment by oblitum** [comment]
Ah, regarding those status messages, I guess I don't have them because I need to put `coc#status()` in my `statusline`, which I didn't yet.

**Comment by oblitum** [comment]
Confirming the above, after adding it to my statusline, it gets stuck with `⠸ Downloading TexLab Server v1.3.0`.

**Comment by oblitum** [comment]
@fannheyward I meant that it does not download :-/ it gets stuck in that message forever, which confirms the behavior of the logs where it doesn't go past 1%.

**Comment by fannheyward** [comment]
What's your OS platform? While downloading, extension will do a pipe work,`extract`, which use `tar` to extract tarball archive.

Does `:messages` prints any error?

**Comment by oblitum** [comment]
I'm on ArchLinux. There's no error in `:messages`.

**Comment by oblitum** [comment]
This issue seems more involved than I thought.

I was able to update it now, trying it again. But it seems that it worked because I opened `nvim` with an empty/nonexisting tex file (`nvim foo.tex`). Then the downloader was able to finish with success as expected.

But it still got stuck when I tried the same in my former test, where I was with complex latex file that I was editing open. I can't release this file, but when I try to update with it open, the downloader gets stuck.
